fhir-tools.sh

```diff
@@ -0,0 +1,243 @@
+
+# Default hapi local server
+DEFAULT_SERVER_URL="http://localhost:8080/fhir"
+
+# Color codes for better readability
+GREEN='\033[0;32m'
+BLUE='\033[0;34m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+
+# Print section header
+section() {
+  echo -e "\n${BLUE}==== $1 ====${NC}"
+}
+
+# Print success message
+success() {
+  echo -e "${GREEN}✓ $1${NC}"
+}
+
+# Print info message
+info() {
+  echo -e "${YELLOW}ℹ $1${NC}"
+}
+
+# Show help
+show_help() {
+  echo "FHIR Resources Tools"
+  echo ""
+  echo "Usage: ./fhir-tools.sh COMMAND [OPTIONS]"
+  echo ""
+  echo "Commands:"
+  echo "  bundle              Generate the auto-bundle.fsh file"
+  echo "  sushi               Run Sushi to compile all FSH files"
+  echo "  post [SERVER_URL]   Post the bundle to a FHIR server (default: $DEFAULT_SERVER_URL)"
+  echo "  hapi-up             Start HAPI FHIR server on Docker (http://localhost:8080)"
+  echo "  hapi-stop           Stop the HAPI FHIR server on Docker"
+  echo "  clean               Remove generated files (bundle and compiled JSON files)"
+  echo "  deploy [SERVER_URL] Generate, compile, and post the bundle (starts HAPI FHIR server if posting locally)"
+  echo "  help                Show this help message"
+  echo ""
+  echo "Examples:"
+  echo "  ./fhir-tools.sh bundle                     # Generate auto-bundle.fsh file"
+  echo "  ./fhir-tools.sh post                       # Post to default server (http://localhost:8080/fhir)"
+  echo "  ./fhir-tools.sh post http://example.com/fhir  # Post to specific server"
+  echo "  ./fhir-tools.sh deploy                     # Full deployment to local server"
+  echo ""
+}
+
+# Generate auto-bundle from all FSH files
+generate_bundle() {
+  section "Generating Bundle"
+  node bundle-generator.js
+  if [ $? -eq 0 ]; then
+    success "Bundle generated successfully"
+  else
+    echo "Error generating bundle"
+    exit 1
+  fi
+}
+
+# Compile FSH files with Sushi
+run_sushi() {
+  section "Compiling FSH Files with Sushi"
+  sushi .
+  if [ $? -eq 0 ]; then
+    success "FSH files compiled successfully"
+  else
+    echo "Error compiling FSH files"
+    exit 1
+  fi
+}
+
+# Post bundle to FHIR server
+post_bundle() {
+  local server_url=${1:-$DEFAULT_SERVER_URL}
+  section "Posting Bundle to $server_url"
+  
+  # Check if the bundle file exists
+  if [ ! -f "fsh-generated/resources/Bundle-auto-compiled-bundle.json" ]; then
+    echo "Bundle file not found. Please generate and compile the bundle first."
+    echo "Run: ./fhir-tools.sh bundle && ./fhir-tools.sh sushi"
+    exit 1
+  fi
+  
+  # Post the bundle to the FHIR server
+  curl -X POST -H "Content-Type: application/fhir+json" \
+    -d @fsh-generated/resources/Bundle-auto-compiled-bundle.json \
+    "$server_url"
+  
+  if [ $? -eq 0 ]; then
+    echo ""
+    success "Bundle posted successfully to $server_url"
+  else
+    echo ""
+    echo "Error posting bundle to $server_url"
+    exit 1
+  fi
+}
+
+# Start HAPI FHIR server on Docker locally
```

Comment:
this should be in a separate script. the main goal of this script is to upload the resources to a FHIR server. If we want to test this with hapi using docker it should be in a different file.